pysimplesoap/server.py

```diff
@@ -384,6 +380,10 @@ def parse_element(name, values, array=False, complex=False):
                         n = "%s%s" % (name, k)
                         parse_element(n, v.items(), complex=True)
                         t = "tns:%s" % n
+                    elif v is None:
```

Comment:
Just a note - `None` could be a key in the `TYPE_MAP`, so I would not change the order here.

Reply:
When the 'v' is dict or list, it occurred a TypeError while looking for key of TYPE_MAP. And I mentioned, before testing for TYPE_MAP, it should test if it is dict type or list type.

Reply:
I just deleted my reply, since I was misunderstanding you've mentioned. Thanks for your review.

Reply:
No problem - I added pull request #165, where you can see what I meant. It is exactly like this pull request, but without reordering the non-problematic checks.